all-contacts/1-pdb-generation/fah-pdbs-make.pl

```diff
@@ -95,41 +103,113 @@ sub generate_pdbs_from_logfile {
 }
 
 sub generate_all_pdbs {
-    my @runs = `ls | grep RUN`;
-    if (scalar(@runs) == 0) { return; }
+    chdir $Project_Dir;
+    my $cwd = getcwd();
```

Comment:
I'm sure you tested this but seems as though there could be unexpected behavior from this subroutine. Why not pass the cwd as an argument?

Reply:
This applies to other subroutines within this script as well.

Reply:
I opted to use Cwd::getcwd because it abstracts away from calling the system's `pwd`. But you are right in that some subroutines can take cwd as an argument. I'll modify that.

Reply:
Ok, I guess we'll start a review for these changes?

Reply:
I think this is related to your comment above regarding using absolute vs relative path, correct? I'd like to keep the logic as is. What I've changed was to add the logic for detecting the correct xtc file to use in `trjconv`.

Reply:
Actually this is a different issue. This concerns calling `pwd` within a subroutine instead of passing the cwd into the subroutine.

Reply:
[`Cwd::getcwd()`](https://perldoc.perl.org/Cwd.html) is recommended over calling `pwd` for portability.

Are you talking about the `generate_all_pdbs()` subroutine?

Reply:
^ is what I mean haha (& nice to know!)

& Yes. I am suggesting calling `getcwd` prior to using the `generate_all_pdbs` subroutine and passing this value into it.
Ex:
```
$cwd = getcwd();
generate_all_pdbs($cwd);
```

Reply:
Above applies to all subroutines that call `getcwd` within them an not passed as an argument.